src/SI_Toolkit/computation_library.py

```diff
@@ -158,14 +161,16 @@ class NumpyLibrary(ComputationLibrary):
     dot = np.dot
     stop_gradient = lambda x: x
     assign = LibraryHelperFunctions.set_to_value
-
+    nan = np.nan
+    isnan=np.isnan
+    string=str
 
 class TensorFlowLibrary(ComputationLibrary):
     lib = 'TF'
     reshape = tf.reshape
     permute = tf.transpose
     newaxis = tf.newaxis
-    shape = lambda x: x.get_shape()  # .as_list()
+    shape = tf.shape #  tobi does not understand reason for this previous definition: # lambda x: x.get_shape()  # .as_list()
```

Comment:
No difference: If `x` is a `tf.Tensor`, then `x.shape` and `x.get_shape()` are identical https://www.tensorflow.org/api_docs/python/tf/Tensor#get_shape

Using `tf.shape(x)` seems to be the best choice here, which is a third option.
From https://www.tensorflow.org/api_docs/python/tf/shape:
tf.shape and Tensor.shape should be identical in eager mode. Within tf.function or within a compat.v1 context, not all dimensions may be known until execution time. Hence, when defining custom layers and models for graph mode, prefer the dynamic tf.shape(x) over the static x.shape.